ethereum-contracts/README.md

```diff
@@ -1,72 +1,61 @@
+2. Interface with the Relayer service, which is used by validators to listen for contract events which are signed and submitted to the Cosmos network as proof of transaction.
+3. Successfully end-to-end test the Cosmos Ethereum Bridge, sending Ethereum and ERC20 tokens from Ethereum to Cosmos.
 
-1) **Testrpc**:
-    - `$ cd ethereum-contracts`
-    - Delete *build* folder: `$ rm -r build/`
-    - `$ truffle migrate --reset --compile-all`
-    - `$ testrpc`
-    - `$ truffle test` on a separate tab
+### Non-Goals of the Smart Contracts
+1. Creating a production-grade system for cross-chain value transfers which enforces strict permissions and limits access to locked funds.
+2. Implementing a validator set which enables observers to submit proof of fund locking transactions on Cosmos to Peggy. These features are not required for unidirectional transfers from Ethereum to Cosmos and will be re-integrated during phase two of the project, which aims to send funds from Cosmos back to Ethereum.
+3. Fully gas optimize and streamline operational functionality; ease and clarity of testing has been favored over some gas management and architectural best practices.
 
+## Ethereum Cosmos Bridge Architecture
+Unidirectional Peggy focuses on core features for unidirectional transfers. This prototype includes functionality to safely lock and unlock Ethereum and ERC20 tokens, emitting associated events which provide the Cosmos based Relayer and Oracle modules with the information required to complete unidirectional transfers. Validators witness the events and submit proof in the form of signed hashes to the Cosmos based modules, which are responsible for aggregating and tallying the Validators’ signatures and their respective signing power. The system is managed by the contract's deployer, designated internally as the relayer, a trusted third-party which can unlock funds and return them their original sender. If the contract’s balances under threat, the relayer can pause the system, temporarily preventing users from depositing additional funds.
 
-2) **Ganache**:
-    - `$ cd ethereum-contracts`
-    - Delete *build* folder: `$ rm -r build/`
-    - `$ truffle migrate --reset --compile-all`
-    - Open Ganache app, open the *Settings* button and make sure it's running on port **_7545_**.
-    - `$ truffle test --network ganache` on a separate tab
+The Peggy Smart Contract is deployed on the Ropsten testnet at address: 0x3de4ef81Ba6243A60B0a32d3BCeD4173b6EA02bb
 
+### Architecture Diagram
+![peggyarchitecturediagram](https://user-images.githubusercontent.com/15370712/58388886-632c7700-7fd9-11e9-962e-4e5e9d92c275.png)
 
-![alt text](./img/ganache_setup.png "Ganache Setup")
+### System Process:
+1. Users lock Ethereum or ERC20 tokens on the Peggy contract, resulting in the emission of an event containing the created item's original sender's Ethereum address, the intended recipient's Cosmos address, the type of token, the amount locked, and the item's unique nonce.
```

Comment:
"created item" is not clear - what "item" is being referred to? What is the nonce? Is this a sort of packet?

Reply:
Yes, an Item is a struct containing information about a unique incoming transaction which executes the `lock()` function. Due to confusion about the name, we should rename the struct to something more descriptive. How about 'LockedTokens' or 'Deposit'?


The nonce is a global variable which protects Peggy against possible replay attacks so that an item's funds can only be unlocked once. Consider this attack scenario:
- An adversary locks X ETH into the contract (move from Ethereum to Cosmos Zone).
- He then requests the reverse operation (move from Cosmos Zone to Ethereum).
- After collecting the validator signatures, the relayer creates a transaction calling the `unlock` function for the amount X to sent to the user.
- The adversary observes the transaction on the blockchain, and copies the signatures
- The adversary, having already received X ETH, creates another transaction calling `unlock` directly himself for the amount X and replaying the captured signatures.
- The amount X is transferred to the adversary again
- The attacker can repeat this attack until the contract has been drained of ETH.

Reply:
`Deposit` sounds great - and a nonce makes sense for replay prevention, agreed.